datalad_fuse/fuse_.py

```diff
@@ -126,34 +126,48 @@ def getattr(self, path, fh=None):
             else:
                 iadok = is_annex_dir_or_key(path)
                 if iadok is not None:
-                    topdir, dir_or_key = iadok
-                    if dir_or_key == "key":
-                        # needs to be open but it is a key. We will let fsspec
-                        # to handle it
-                        pass
-                    elif dir_or_key == "dir":
+                    if isinstance(iadok, AnnexKey):
+                        if iadok.size is not None:
+                            lgr.debug("Got size from key")
+                            r = mkstat(
+                                is_file=True,
+                                size=iadok.size,
+                                timestamp=self._adapter.get_commit_datetime(path),
```

Comment:
aren't we under `.git/annex/objects` here and thus the commit date wouldn't really be pertinent to that key file ?  then let's just use some arbitrary timestamp -- e.g. fixed timestamp on when we started this fusefs instance.  Should help us to save some cpu cycles

Reply:
The commit date is cached when the adapter for the (sub)dataset is created [[link](https://github.com/datalad/datalad-fuse/blob/5421d3a1e0f43e09947964a5c5cc4a2bb78584d0/datalad_fuse/fsspec.py#L35)], so there aren't many cycles to save.

Reply:
it still needs to do some traversal to figure out the top of the dataset right?  indeed might be negligible though

Reply:
ok, let's proceed for now as is, and optimize if we see it adds penalty